### ISSUE TITLE
problems with PISCES variables in EC-Earth-CC

### ISSUE THREAD

**Issue by etiennesky** [EC-Earth/ece2cmor3#521]
I am running from the c-cycle branch (updated with trunk) and ece2cmor3 v 1.2.0 with the CMIP/EC-EARTH-CC/cmip6-experiment-CMIP-piControl ctrl files.

ece2cmor complained about the dissicnatos variable

```
! Error: You defined variable 'dissicnatos' (table Omon) with axis id 'depth_coord' which is not part of this variable according to your table, it says: ( time latitude longitude )
!
```
The problem is that in the file_def it is defined as a 3d var, but in the Omon table as a 2d var.

I modified the CMIP6_Omon.json file to add the olevel dimension for dissicnatos , and cmorization continued.

I have the same issue with intdoc and talknatos - adding olevel to the Omon table for those vars also fixed it...

But it this the proper fix ?

**Comment by etiennesky** [comment]
@pabretonniere @goord @treerink @aearamos

**Comment by etiennesky** [comment]
Looking better into the field_def and file_defs..

I think the problem is in the file_defs, not the Omon tables.

dissicnatos is defined  in Omon  as "Surface Natural Dissolved Inorganic Carbon Concentration"

but in file_def file_def_nemo-pisces.xml dissicnatos is mapped to DIC ("Dissolved inorganic Concentration")  - a3d var -    whereas it should be mapped to something like  DICSFC_E3T ("Surface Dissolved inorganic Content"     ) which is a 2d var, but this is already mapped correctly to dissicos.  

I don't see any var which matches "Surface Natural Dissolved Inorganic Carbon Concentration"  in the field_def.


Same issue with talknatos... there is no variable for "Surface Natural Total Alkalinity", only AlkaliniSFC_E3T ("Surface Total Alkalinity Content" ).

And intdoc is defined as long_name": "Dissolved Organic Carbon Content",  "comment": "Vertically integrated DOC (explicit pools only)",  I don't see a variable for the vertical sum of DOC.

Perhaps we should ignore these variables?

**Comment by etiennesky** [comment]
Another problem is that the _pisces_grid_T_SFC.nc file contains the proper 2D variables (dissicos, talkos), but this file is not being copied into output/nemo/??? at the end of the runscript.  Bu copying the file from the runtime, I was able to cmorize them fine.  Will commit this change to the ec-earth c-cycle branch.

After removing from the varlist.json the variables dissicnatos talknatos and intdoc , I was able to finish the cmorization.

**Comment by etiennesky** [comment]
Hi, will have a look into this next week.

**Comment by treerink** [comment]
I had a first look now. I agree that changing the tables is not the way to go, and indeed the definition of `dissicnatos` [here](http://clipc-services.ceda.ac.uk/dreq/u/c96c470c-c5f0-11e6-ac20-5404a60d96b5.html) says it is a surface field. Indeed currently it is linked to
```xml
field_ref="DIC"                                    grid_ref="grid_T_3D" 
```
which obviously doesn't look correct: 3D instead of 2D. It comes from the Shaconemo ping file `ping_ocnBgChem_DR1.00.27.xml` in our ec-earth repository in the ctrl directory. Note that it concerns a variable which had the dummy status (i.e. unidentified) in the Shaconemo repository, but that it has been added by us on top of the Shaconemo vendor version, see [ece porta issuel 591](https://dev.ec-earth.org/issues/595#note-1). The table says "DIC in upper layer", so apparently filtering only the upper layer has not been done. Could this easily be done with an expression in the ping file (which then is shipped into the file_def file)?

Also the variables `intdoc` and `talknatos` are present in the table of this  [ece porta issuel 591](https://dev.ec-earth.org/issues/595#note-1) post.

The table comes from [this earlier discussion](https://dev.ec-earth.org/issues/518#note-145), from there it is possible to trace back the identification [doc](https://docs.google.com/spreadsheets/d/189ClDiFx3urHpiuXuMnQKkkpIeV2P-2Zy11ffNnQ408/edit#gid=1234946926) from one can see that Raffaele seems the one with the most knowledge of this.

**Comment by etiennesky** [comment]
Hi @treerik sorry for not answering you earlier.

This is also very important for the OMIP runs which have the same problem.

In a nutshell, we have to make the definition for dissicnatos the same as that for dissicos.
Also talknatos has to be the same as talkos.

I think by editing the ping files it should be fine.

Also, we can add the info for related variable dissicnat - same as dissic.

Another point is that the "nat" vs "no-nat" flavours should be activated depending if you are doing a piControl of historical/future run : 

dissicnatos : Dissolved inorganic carbon (CO3+HCO3+H2CO3) concentration at preindustrial atmospheric xCO2
dissicos : Dissolved inorganic carbon (CO3+HCO3+H2CO3) concentration

Finally, since intdoc is a priority 2 we should just make sure it is not output, this would fix the issue for us.

Let me know if you need any feedback, perhaps if you generate the file-def for the C4MIP piControl and historical after making the changes we can check the generated file-defs.

**Comment by etiennesky** [comment]
Hi, just a heads up to say that now that the c-cycle branch is merged into trunk.

**Comment by etiennesky** [comment]
Hi I just talked to Rafa, and for now, the quick & dirty solution is to disable the 3 variables (dissicnatos, talknatos and intdoc), and in the near future we will have to fix these and other variables.

For OMIP the *nat and *natos are important, but for c4mip we cannot deliver them so we won't use them.

Here is what we will need to do in practical terms in the ping file:

- CMIP6_dissicnatos should have the same definition as CMIP6_dissicos 
- CMIP6_dissicnat should have the same definition as CMIP6_dissic (it is currently dummy)
- CMIP6_talknatos should have the same definition as CMIP6_talkos
- CMIP6_talknat should have the same definition as CMIP6_talk
- and other stuff yet undecided.

**Comment by treerink** [comment]
Ok I locally made this changes in my ping files, let's see into what issues I will be running with that. Otherwise I will opt for the quick solution, by just excepting these variables for now from the json data request files.

**Comment by treerink** [comment]
I am changing the following in the `ping_ocnBgChem_DR1.00.27.xml` file (to be committed):
> * CMIP6_dissicnatos should have the same definition as CMIP6_dissicos

Both `field_ref` and `expression` attributes of `dissicnatos` equal those of `dissicos` now.

> * CMIP6_dissicnat should have the same definition as CMIP6_dissic   **(it is currently dummy)**

Both `field_ref` and `expression` attributes of `dissicnat` equal those of `dissic` now.

> * CMIP6_talknatos should have the same definition as CMIP6_talkos

Both `field_ref` and `expression` attributes of `talknatos` equal those of `talkos` now.

> * CMIP6_talknat should have the same definition as CMIP6_talk          **(it is currently dummy)**

Both `field_ref` and `expression` attributes of `talknat` equal those of `talk` now.


Resulting in the following `svn diff`:
```diff
--- ping_ocnBgChem_DR1.00.27.xml	(revision 7391)
+++ ping_ocnBgChem_DR1.00.27.xml	(working copy)
@@ -74,8 +74,8 @@
    <field id="CMIP6_dissic"        field_ref="DIC_E3T"  expr="@DIC_E3T / @e3t * 1e-3" > DIC_E3T / e3t * 1e-3 </field> <!-- P1 (mol m-3) mole_concentration_of_dissolved_inorganic_carbon_in_sea_water : Dissolved inorganic carbon (CO3+HCO3+H2CO3) concentration -->
    <field id="CMIP6_dissicabio"    field_ref="dummy_XYO"        /> <!-- P1 (mol m-3) mole_concentration_of_dissolved_inorganic_carbon_abiotic_analogue_in_sea_water : Abiotic Dissolved inorganic carbon (CO3+HCO3+H2CO3) concentration -->
    <field id="CMIP6_dissicabioos"  field_ref="dummy_XY"         /> <!-- P1 (mol m-3) mole_concentration_of_dissolved_inorganic_carbon_abiotic_analogue_in_sea_water : Abiotic Dissolved inorganic carbon (CO3+HCO3+H2CO3) concentration -->
-   <field id="CMIP6_dissicnat"     field_ref="dummy_XYO"        /> <!-- P1 (mol m-3) mole_concentration_of_dissolved_inorganic_carbon_natural_analogue_in_sea_water : Dissolved inorganic carbon (CO3+HCO3+H2CO3) concentration at preindustrial atmospheric xCO2 -->
-   <field id="CMIP6_dissicnatos"   field_ref="DIC"         /> <!-- P1 (mol m-3) mole_concentration_of_dissolved_inorganic_carbon_natural_analogue_in_sea_water : Dissolved inorganic carbon (CO3+HCO3+H2CO3) concentration at preindustrial atmospheric xCO2 -->
+   <field id="CMIP6_dissicnat"     field_ref="DIC_E3T"  expr="@DIC_E3T / @e3t * 1e-3" > DIC_E3T / e3t * 1e-3 </field> <!-- P1 (mol m-3) mole_concentration_of_dissolved_inorganic_carbon_natural_analogue_in_sea_water : Dissolved inorganic carbon (CO3+HCO3+H2CO3) concentration at preindustrial atmospheric xCO2 -->
+   <field id="CMIP6_dissicnatos"   field_ref="DICSFC_E3T"  expr="@DICSFC_E3T / @E3TSFC * 1e-3" >  DICSFC_E3T / E3TSFC * 1e-3 </field> <!-- P1 (mol m-3) mole_concentration_of_dissolved_inorganic_carbon_natural_analogue_in_sea_water : Dissolved inorganic carbon (CO3+HCO3+H2CO3) concentration at preindustrial atmospheric xCO2 -->
    <field id="CMIP6_dissicos"      field_ref="DICSFC_E3T"  expr="@DICSFC_E3T / @E3TSFC * 1e-3" >  DICSFC_E3T / E3TSFC * 1e-3 </field> <!-- P1 (mol m-3) mole_concentration_of_dissolved_inorganic_carbon_in_sea_water : Dissolved inorganic carbon (CO3+HCO3+H2CO3) concentration -->
    <field id="CMIP6_dissoc"        field_ref="DOC_E3T"     expr="@DOC_E3T / @e3t * 1e-3" >  DOC_E3T / e3t * 1e-3 </field>  <!-- P1 (mol m-3) mole_concentration_of_dissolved_organic_carbon_in_sea_water : Sum of dissolved carbon component concentrations explicitly represented (i.e. not ~40 uM refractory unless explicit) -->
    <field id="CMIP6_dissocos"        field_ref="DOCSFC_E3T"     expr="@DOCSFC_E3T / @E3TSFC * 1e-3" >  DOCSFC_E3T / E3TSFC * 1e-3 </field> <!-- P2 (mol m-3) mole_concentration_of_dissolved_organic_carbon_in_sea_water : Sum of dissolved carbon component concentrations explicitly represented (i.e. not ~40 uM refractory unless explicit) -->
@@ -222,8 +222,8 @@
    <field id="CMIP6_spco2abio"     field_ref="dummy_XY"         /> <!-- P1 (Pa) surface_carbon_dioxide_abiotic_analogue_partial_pressure_difference_between_sea_water_and_air : The surface called "surface" means the lower boundary of the atmosphere. The chemical formula for carbon dioxide is CO2. In ocean biogeochemistry models, an "abiotic analogue" is used to simulate the effect on a modelled variable when biological effects on ocean carbon concentration and alkalinity are ignored. The partial pressure of a gaseous constituent of air is the pressure which it alone would exert with unchanged temperature and number of moles per unit volume. The partial pressure of a dissolved gas in sea water is the partial pressure in air with which it would be in equilibrium. The partial pressure difference between sea water and air is positive when the partial pressure of the dissolved gas in sea water is greater than the partial pressure in air. -->
    <field id="CMIP6_spco2nat"      field_ref="dummy_XY"         /> <!-- P1 (Pa) surface_carbon_dioxide_natural_analogue_partial_pressure_difference_between_sea_water_and_air : The surface called "surface" means the lower boundary of the atmosphere. The chemical formula for carbon dioxide is CO2. In ocean biogeochemistry models, a "natural analogue" is used to simulate the effect on a modelled variable of imposing preindustrial atmospheric carbon dioxide concentrations, even when the model as a whole may be subjected to varying forcings. The partial pressure of a gaseous constituent of air is the pressure which it alone would exert with unchanged temperature and number of moles per unit volume. The partial pressure of a dissolved gas in sea water is the partial pressure in air with which it would be in equilibrium. The partial pressure difference between sea water and air is positive when the partial pressure of the dissolved gas in sea water is greater than the partial pressure in air. -->
    <field id="CMIP6_talk"          field_ref="Alkalini_E3T" expr="@Alkalini_E3T / @e3t * 1e-3" > Alkalini_E3T / e3t * 1e-3 </field>  <!-- P1 (mol m-3) sea_water_alkalinity_expressed_as_mole_equivalent : total alkalinity equivalent concentration (including carbonate, borate, phosphorus, silicon, and nitrogen components) -->
-   <field id="CMIP6_talknat"       field_ref="dummy_XYO"        /> <!-- P1 (mol m-3) sea_water_alkalinity_natural_analogue_expressed_as_mole_equivalent : total alkalinity equivalent concentration (including carbonate, borate, phosphorus, silicon, and nitrogen components) at preindustrial atmospheric xCO2 -->
-   <field id="CMIP6_talknatos"     field_ref="Alkalini"         /> <!-- P1 (mol m-3) sea_water_alkalinity_natural_analogue_expressed_as_mole_equivalent : total alkalinity equivalent concentration (including carbonate, borate, phosphorus, silicon, and nitrogen components) at preindustrial atmospheric xCO2 -->
+   <field id="CMIP6_talknat"       field_ref="Alkalini_E3T" expr="@Alkalini_E3T / @e3t * 1e-3" > Alkalini_E3T / e3t * 1e-3 </field> <!-- P1 (mol m-3) sea_water_alkalinity_natural_analogue_expressed_as_mole_equivalent : total alkalinity equivalent concentration (including carbonate, borate, phosphorus, silicon, and nitrogen components) at preindustrial atmospheric xCO2 -->
+   <field id="CMIP6_talknatos"     field_ref="AlkaliniSFC_E3T" expr="@AlkaliniSFC_E3T / @E3TSFC * 1e-3" > AlkaliniSFC_E3T / E3TSFC * 1e-3 </field> <!-- P1 (mol m-3) sea_water_alkalinity_natural_analogue_expressed_as_mole_equivalent : total alkalinity equivalent concentration (including carbonate, borate, phosphorus, silicon, and nitrogen components) at preindustrial atmospheric xCO2 -->
    <field id="CMIP6_talkos"          field_ref="AlkaliniSFC_E3T" expr="@AlkaliniSFC_E3T / @E3TSFC * 1e-3" > AlkaliniSFC_E3T / E3TSFC * 1e-3 </field> <!-- P1 (mol m-3) sea_water_alkalinity_expressed_as_mole_equivalent : total alkalinity equivalent concentration (including carbonate, borate, phosphorus, silicon, and nitrogen components) -->
    <field id="CMIP6_zmeso"         field_ref="ZOO2_E3T" expr="@ZOO2_E3T / @e3t * 1e-3" > ZOO2_E3T / e3t * 1e-3 </field>  <!-- P1 (mol m-3) mole_concentration_of_mesozooplankton_expressed_as_carbon_in_sea_water : carbon  concentration from mesozooplankton (20-200 um) component alone -->
    <field id="CMIP6_zmesoos"       field_ref="ZOO2SFC_E3T" expr="@ZOO2SFC_E3T / @E3TSFC * 1e-3" > ZOO2SFC_E3T / E3TSFC * 1e-3 </field>   <!-- P2 (mol m-3) mole_concentration_of_mesozooplankton_expressed_as_carbon_in_sea_water : carbon  concentration from mesozooplankton (20-200 um) component alone -->
```

**Comment by etiennesky** [comment]
Hi @treerink thanks forthis. Also please make sure that intdoc is not in the file_def files, because its definition is wrong as I wrote above, and since it is a priority 2 we don't want it.

**Comment by treerink** [comment]
Yes `intdoc` will be added to the omit list in `drq2varlist.py`, so it won't sow up in the json data request files.

I am now processing the pisces changes for the identification process and will be rerunning `genecec` with that.

**Comment by treerink** [comment]
All changes uploaded in the `r7144-control-output-files` branch.

Closing this issue.

**Comment by etiennesky** [comment]
Hi @treerink I don't see any changes in the various file_def_nemo-pisces.xml files in that branch - only the ping file as well as some .json files have been updated.

**Comment by treerink** [comment]
Thanks Etienne for spotting this.

It looks like I forgot to apply `create-basic-ec-earth-cmip6-nemo-namelist.py` before running `genecec` after I made all the changes. Probably at some stage I reverted the changes, while working on different issues at the same time.

I will run now `genecec` and see if that gives the expected changes. If so I will open a new branch as the last one has been merged already.
